test/volume_test.py

```diff
@@ -367,3 +372,22 @@ async def test_open_files_error_annotation(tmp_path):
 def test_invalid_name(servicer, client, name):
     with pytest.raises(InvalidError, match="Invalid Volume name"):
         modal.Volume.lookup(name)
+
+
+def test_warn_on_local_volume_mount(client, servicer):
+    app = modal.App()
+    vol = modal.Volume.from_name("xyz")
+
+    assert modal.is_local() == True
+    with pytest.warns(match="local"):
+        app.function(volumes={"/foo": vol})(dummy).local()
+
+
+def test_no_warn_on_remote_local_volume_mount(client, servicer, recwarn):
+    _run_container(
+        servicer,
+        "test.supports.functions",
+        "volume_func_outer",
+        inputs=_get_inputs(((), {})),
+    )
+    assert len(recwarn) == 0
```

Comment:
Alternatively, this could go in `container_test` so we don't need so much rearrangement in this PR and preserve more of the `git blame`?